examples/experiment_web_search.py

```diff
@@ -0,0 +1,160 @@
+import os
+import re
+from typing import Dict, Any, List
+
+from haystack import Document
+from haystack.agents import Agent, Tool
+from haystack.nodes import PromptNode, PromptTemplate
+from haystack.nodes.search_engine import WebSearch, NeuralWebSearch
+
+search_key = os.environ.get("SERPERDEV_API_KEY")
+if not search_key:
+    raise ValueError("Please set the SERPERDEV_API_KEY environment variable")
+
+ws = WebSearch(api_key=search_key)
+
+pn = PromptNode("text-davinci-003", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
+
+prompt_text = """
+Answer the following question using the paragraphs below as sources. An answer should be short, a few words at most.
+Provide the answer as the last generated line starting with Answer:
+
+Paragraphs: $paragraphs
+
+Question: $question
+
+Instructions: Consider all the paragraphs above and their corresponding scores to generate the answer. While a single
+paragraph may have a high score, it's important to consider all paragraphs for the same answer candidate to answer
+accurately.
+
+Let's think step-by-step, we have the following distinct answer possibilities:
+
+"""
+pt = PromptTemplate("neural_web_search", prompt_text=prompt_text)
```

Comment:
Yeah we could. I didn't think to include NeuralWebSearch as a component - because it is a convenience component to start experimenting but it could be a useful example. And others can customize it.